functions/Get-DbaUserPermission.ps1

```diff
@@ -215,17 +215,65 @@ function Get-DbaUserPermission {
                 $dbs = $dbs | Where-Object IsSystemObject -eq $false
             }
 
-            #reset $serverDT
-            $serverDT = $null
+            # grab server data (just the one time):
+            Write-Message -Level Verbose -Message "Building data table for server objects"
 
-            foreach ($db in $dbs) {
-                Write-Message -Level Verbose -Message "Processing $db on $instance"
+            $sqlFile = Join-Path -Path $script:PSModuleRoot -ChildPath "bin\stig.sql"
+            $sql = [System.IO.File]::ReadAllText("$sqlFile")
+            $sql = $sql.Replace("<TARGETDB>", $master.Name)
 
-                $db.ExecuteNonQuery($endSQL)
+            try {
+                $master.ExecuteNonQuery($sql)
+            } catch {
+                # here to avoid an empty catch
+                $null = 1
+            } # sometimes it complains about not being able to drop the stig schema if the person Ctrl-C'd before.
 
-                if ($db.IsAccessible -eq $false) {
-                    Stop-Function -Message "The database $db is not accessible" -Continue
+            try {
+                $serverDT = $master.Query($serverSQL)
```

Comment:
Why do you execute this query against master? Why not tempdb?

Reply:
Personal preference. It now executes against tempdb (see conversation further up in the code).